x-pack/plugins/uptime/common/rules/uptime_rule_field_map.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const uptimeRuleFieldMap = {
+  // common fields
+  'monitor.id': {
+    type: 'keyword',
+  },
+  'url.full': {
+    type: 'keyword',
+  },
+  'observer.geo.name': {
+    type: 'keyword',
+  },
+  reason: {
+    type: 'text',
+  },
+  // monitor status alert fields
+  'error.message': {
+    type: 'text',
+  },
+  'agent.name': {
+    type: 'keyword',
+  },
+  'monitor.name': {
+    type: 'keyword',
+  },
+  'monitor.type': {
+    type: 'keyword',
+  },
+  // tls alert fields
+  'tls.server.x509.issuer.common_name': {
+    type: 'keyword',
+  },
+  'tls.server.x509.subject.common_name': {
+    type: 'keyword',
+  },
+  'tls.server.x509.not_after': {
+    type: 'date',
+  },
+  'tls.server.x509.not_before': {
+    type: 'date',
+  },
+  'tls.server.hash.sha256': {
+    type: 'keyword',
+  },
+  // anomaly alert fields
+  'anomaly.start': {
```

Comment:
do we need more fields for anomaly?

Reply:
the majority of the anomaly fields have been transitioned to use RAC technical fields, as well as ecs fields listed in the `common` section. You can see the change in the `anomaly_duration.ts` server file.